proteinbenchmark/protein_system.py

```diff
@@ -146,7 +153,7 @@ def setup(self):
                 )
 
         # Solvate, add ions, and construct OpenMM system
-        if not exists_and_not_empty(self.openmm_system):
+        if (self.sim_platform != 'gmx' and not exists_and_not_empty(self.openmm_system)) or (self.sim_platform == 'gmx' and not exists_and_not_empty(f'{self.setup_prefix}.top')):
```

Comment:
This logic is a bit complicated. We should probably replace the `self.openmm_system` attribute with something like `self.parametrized_system` that contains the System XML for OpenMM or the top file for GMX.